### PR TITLE
refactor: cobra.flux_analysis.deletion

### PR DESCRIPTION
**Pull request by synchon**
* [ ] fix #(issue number)
* [x] description of feature/fix
* [x] tests added/passed
* [ ] add an entry to the [next release](../release-notes/next-release.md)

This PR adds typing annotations for `cobra.flux_analysis.deletion` and its unit tests. It also upgrades the code style for them to maintain Python 3.6+ compatibility.

### CODE REVIEW

File: src/cobra/flux_analysis/deletion.py [hunk 1]
```diff
@@ -1,21 +1,41 @@
+"""Provide functions for reactions and gene deletions."""
```
Comment (Midnighter) [inline src/cobra/flux_analysis/deletion.py]:
```suggestion
"""Provide functions for reaction and gene deletions."""
```

File: src/cobra/flux_analysis/deletion.py [hunk 2]
```diff
@@ -153,50 +240,85 @@ def extract_knockout_results(result_iter):
...
     return result
 
 
 def single_reaction_deletion(
-    model, reaction_list=None, method="fba", solution=None, processes=None, **kwargs
-):
-    """
-    Knock out each reaction from a given list.
+    model: Model,
+    reaction_list: Optional[List[Reaction]] = None,
```
Comment (Midnighter) [inline src/cobra/flux_analysis/deletion.py]:
```suggestion
    reaction_list: Optional[List[Union[Reaction, str]]] = None,
```

File: src/cobra/flux_analysis/deletion.py [hunk 3]
```diff
@@ -219,34 +341,39 @@ def single_reaction_deletion(
...
     )
 
 
 def single_gene_deletion(
-    model, gene_list=None, method="fba", solution=None, processes=None, **kwargs
-):
-    """
-    Knock out each gene from a given list.
+    model: Model,
+    gene_list: Optional[List[Gene]] = None,
```
Comment (Midnighter) [inline src/cobra/flux_analysis/deletion.py]:
```suggestion
    gene_list: Optional[List[Union[Gene, str]]] = None,
```

File: src/cobra/flux_analysis/deletion.py [hunk 4]
```diff
@@ -269,45 +396,45 @@ def single_gene_deletion(
...
-    method="fba",
-    solution=None,
-    processes=None,
-    **kwargs
-):
-    """
-    Knock out each reaction pair from the combinations of two given lists.
+    model: Model,
+    reaction_list1: Optional[List[Reaction]] = None,
+    reaction_list2: Optional[List[Reaction]] = None,
```
Comment (Midnighter) [inline src/cobra/flux_analysis/deletion.py]:
```suggestion
    reaction_list1: Optional[List[Union[Reaction, str]]] = None,
    reaction_list2: Optional[List[Union[Reaction, str]]] = None,
```

File: src/cobra/flux_analysis/deletion.py [hunk 5]
```diff
@@ -334,45 +460,45 @@ def double_reaction_deletion(
...
-    method="fba",
-    solution=None,
-    processes=None,
-    **kwargs
-):
-    """
-    Knock out each gene pair from the combination of two given lists.
+    model: Model,
+    gene_list1: Optional[List[Gene]] = None,
+    gene_list2: Optional[List[Gene]] = None,
```
Comment (Midnighter) [inline src/cobra/flux_analysis/deletion.py]:
```suggestion
    gene_list1: Optional[List[Union[Gene, str]]] = None,
    gene_list2: Optional[List[Union[Gene, str]]] = None,
```

**Review by Midnighter — CHANGES_REQUESTED** [review]
Just a few type changes, otherwise looks good 👍🏼

**Comment by synchon** [comment]
The latest commit should address your changes. :)

**Review by Midnighter — APPROVED** [review]

**Pull request merged**